packages/elements/src/components/types.ts

```diff
@@ -60,3 +60,8 @@ export type UserInputInterceptor = (value: string) => string;
  * An arbitary string (or close by default) referencing the action of a dialog.
  */
 export type DialogCloseAction = 'close'|string;
+
+export type KeyValue = {
```

Comment:
Well, it is necessary for `value` to be a `string` because that's what the library uses to display the options. For now, I would suggest keeping it that way.